src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
+        long[] divisors = GetDivisors(candidate);
+        int sum = 0;
+        for (int d = 0; d < 1000; d++)
```

Comment:
A test is missing in the for statement, otherwise it will not consider the size of the array.
`for (int d = 0; d < 1000 && d < divisors.length; d++)`